pyadjoint/tape.py

```diff
@@ -722,11 +722,12 @@ def __iter__(self):
         return self
 
     def __next__(self):
+        step = next(self.iterator)
```

Comment:
Why?

Reply:
This is required for dynamic schedules. If `next(self.iterator)` is passed after `self.tape.end_timestep()`, an additional (and not necessary) time-step will be executed because the total timestep is not yet defined. In conclusion, [this line tes L74](https://github.com/dolfin-adjoint/pyadjoint/blob/master/tests/firedrake_adjoint/test_burgers_newton.py#L74)t passes.